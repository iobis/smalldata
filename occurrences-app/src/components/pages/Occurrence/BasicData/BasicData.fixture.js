import BasicData from './BasicData'

export default [{
  component: BasicData
}]
