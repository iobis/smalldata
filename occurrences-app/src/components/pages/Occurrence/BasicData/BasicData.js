import React, { useState, useEffect, useRef } from 'react'
import bulmaCalendar from 'bulma-calendar'

export default function BasicData() {
  const [beginDate, setBeginDate] = useState(new Date())
  const [endDate, setEndDate] = useState(null)

  return (
    <div className="basic-dataset section is-fluid">
      <div className="field">
        <label className="label">Scientific name</label>
        <div className="control">
          <input className="input" type="text" placeholder="Text input"/>
        </div>
      </div>
      <div className="field">
        <label className="label">Event date begin</label>
        <DatePicker onChange={setBeginDate} value={beginDate}/>
      </div>
      <div className="field">
        <label className="label">
          Event date end
        </label>
        <div className="control">
          <DatePicker onChange={setEndDate} value={endDate}/>
        </div>
        <p className="help">optional: only in case of date range</p>
      </div>
      <div className="field">
        <label className="label">Occurrence status</label>
        <div className="control">
          <InputRadio text="absent" name="occurrence-status"/>
          <InputRadio text="present" name="occurrence-status"/>
        </div>
      </div>
      <div className="field">
        <label className="label">Basis of record</label>
        <div className="control">
          <InputRadio text="human observation" name="basis"/>
          <InputRadio text="fossil specimen" name="basis"/>
          <InputRadio text="living specimen" name="basis"/>
          <InputRadio text="machine specimen" name="basis"/>
          <InputRadio text="preserved specimen" name="basis"/>
        </div>
      </div>
      <div className="field">
        <label className="label">
          Sex <small>(optional)</small>
        </label>
        <div className="control">
          <InputRadio text="male" name="sex"/>
          <InputRadio text="female" name="sex"/>
          <InputRadio text="hermaphrodite" name="sex"/>
        </div>
      </div>
      <div className="field">
        <label className="label">
          Lifestage <small>(optional)</small>
        </label>
        <div className="control">
          <InputRadio text="egg" name="lifestage"/>
          <InputRadio text="eft" name="lifestage"/>
          <InputRadio text="juvenile" name="lifestage"/>
          <InputRadio text="adult" name="lifestage"/>
        </div>
      </div>
    </div>
  )
}

function InputRadio({ checked, text, name }) {
  return (
    <>
      <input className="is-checkradio" id={text} type="radio" name={name} checked={checked}/>
      <label htmlFor={text}>{text}</label>
    </>
  )
}

function DatePicker({ onChange, value }) {
  const datePickerEl = useRef()

  useEffect(() => {
    const calendar = bulmaCalendar.attach(datePickerEl.current, {
      type:        'date',
      displayMode: 'default',
      startDate:   value !== null ? new Date(value) : null
    })
    calendar[0].on('select', (e) => onChange(e.data))
  }, [])

  return <input className="input" ref={datePickerEl} type="date" placeholder="Text input"/>
}
