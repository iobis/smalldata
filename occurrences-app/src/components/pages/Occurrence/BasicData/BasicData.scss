.basic-dataset {
  .field {
    .label {
      small {
        font-size: 0.75em;
        font-style: italic;
      }
    }
  }
}
