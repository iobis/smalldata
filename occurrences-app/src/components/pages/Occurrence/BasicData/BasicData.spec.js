import React from 'react'
import renderer from 'react-test-renderer'
import BasicData from './BasicData'

describe('SelectDataset', () => {
  it('renders correctly', () => {
    expect(renderer.create(createComponent()).toJSON()).toMatchSnapshot()
  })
})

function createComponent() {
  return (
    <BasicData
      basicData={{
        basisOfRecord:    'humanObservation',
        beginDate:        new Date('2019-01-02T12:30:40'),
        endDate:          new Date('2020-01-02T12:30:40'),
        lifestage:        'egg',
        occurrenceStatus: 'absent',
        scientificName:   'scientific name',
        sex:              'male'
      }}
      onChange={jest.fn()}/>
  )
}
