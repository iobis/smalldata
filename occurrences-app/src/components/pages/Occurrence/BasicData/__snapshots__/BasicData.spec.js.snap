// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectDataset renders correctly 1`] = `
<div
  className="basic-data section is-fluid"
>
  <div
    className="columns"
  >
    <div
      className="field is-four-fifths column"
    >
      <label
        className="label"
      >
        occurrenceForm.basicData.scientificName
      </label>
      <div
        className="control"
      >
        <input
          className="input"
          onChange={[Function]}
          placeholder="occurrenceForm.basicData.scientificName"
          type="text"
          value="scientific name"
        />
      </div>
    </div>
  </div>
  <div
    className="columns"
  >
    <div
      className="event-begin-date column field is-two-fifths"
    >
      <label
        className="label"
      >
        occurrenceForm.basicData.eventBeginDate
      </label>
      <input
        className="input"
        type="date"
      />
    </div>
    <div
      className="event-end-date column field is-two-fifths"
    >
      <label
        className="label"
      >
        occurrenceForm.basicData.eventEndDate
      </label>
      <div
        className="control"
      >
        <input
          className="input"
          type="date"
        />
      </div>
      <p
        className="help"
      >
        occurrenceForm.basicData.eventEndDateHelp
      </p>
    </div>
  </div>
  <div
    className="input-radio-group field"
  >
    <label
      className="label"
    >
      occurrenceForm.basicData.occurrenceStatus.title
    </label>
    <div
      className="control"
    >
      <input
        checked={true}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.occurrenceStatus-absent"
        name="occurrenceForm.basicData.occurrenceStatus"
        onChange={[Function]}
        type="radio"
        value="absent"
      />
      <label
        htmlFor="occurrenceForm.basicData.occurrenceStatus-absent"
      >
        occurrenceForm.basicData.occurrenceStatus.absent
      </label>
      <input
        checked={false}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.occurrenceStatus-present"
        name="occurrenceForm.basicData.occurrenceStatus"
        onChange={[Function]}
        type="radio"
        value="present"
      />
      <label
        htmlFor="occurrenceForm.basicData.occurrenceStatus-present"
      >
        occurrenceForm.basicData.occurrenceStatus.present
      </label>
    </div>
  </div>
  <div
    className="input-radio-group field"
  >
    <label
      className="label"
    >
      occurrenceForm.basicData.basisOfRecord.title
    </label>
    <div
      className="control"
    >
      <input
        checked={true}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.basisOfRecord-humanObservation"
        name="occurrenceForm.basicData.basisOfRecord"
        onChange={[Function]}
        type="radio"
        value="humanObservation"
      />
      <label
        htmlFor="occurrenceForm.basicData.basisOfRecord-humanObservation"
      >
        occurrenceForm.basicData.basisOfRecord.humanObservation
      </label>
      <input
        checked={false}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.basisOfRecord-fossilSpecimen"
        name="occurrenceForm.basicData.basisOfRecord"
        onChange={[Function]}
        type="radio"
        value="fossilSpecimen"
      />
      <label
        htmlFor="occurrenceForm.basicData.basisOfRecord-fossilSpecimen"
      >
        occurrenceForm.basicData.basisOfRecord.fossilSpecimen
      </label>
      <input
        checked={false}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.basisOfRecord-livingSpecimen"
        name="occurrenceForm.basicData.basisOfRecord"
        onChange={[Function]}
        type="radio"
        value="livingSpecimen"
      />
      <label
        htmlFor="occurrenceForm.basicData.basisOfRecord-livingSpecimen"
      >
        occurrenceForm.basicData.basisOfRecord.livingSpecimen
      </label>
      <input
        checked={false}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.basisOfRecord-machineSpecimen"
        name="occurrenceForm.basicData.basisOfRecord"
        onChange={[Function]}
        type="radio"
        value="machineSpecimen"
      />
      <label
        htmlFor="occurrenceForm.basicData.basisOfRecord-machineSpecimen"
      >
        occurrenceForm.basicData.basisOfRecord.machineSpecimen
      </label>
      <input
        checked={false}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.basisOfRecord-preservedSpecimen"
        name="occurrenceForm.basicData.basisOfRecord"
        onChange={[Function]}
        type="radio"
        value="preservedSpecimen"
      />
      <label
        htmlFor="occurrenceForm.basicData.basisOfRecord-preservedSpecimen"
      >
        occurrenceForm.basicData.basisOfRecord.preservedSpecimen
      </label>
    </div>
  </div>
  <div
    className="input-radio-group field"
  >
    <label
      className="label"
    >
      occurrenceForm.basicData.sex.title
    </label>
    <div
      className="control"
    >
      <input
        checked={true}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.sex-male"
        name="occurrenceForm.basicData.sex"
        onChange={[Function]}
        type="radio"
        value="male"
      />
      <label
        htmlFor="occurrenceForm.basicData.sex-male"
      >
        occurrenceForm.basicData.sex.male
      </label>
      <input
        checked={false}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.sex-female"
        name="occurrenceForm.basicData.sex"
        onChange={[Function]}
        type="radio"
        value="female"
      />
      <label
        htmlFor="occurrenceForm.basicData.sex-female"
      >
        occurrenceForm.basicData.sex.female
      </label>
      <input
        checked={false}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.sex-hermaphrodite"
        name="occurrenceForm.basicData.sex"
        onChange={[Function]}
        type="radio"
        value="hermaphrodite"
      />
      <label
        htmlFor="occurrenceForm.basicData.sex-hermaphrodite"
      >
        occurrenceForm.basicData.sex.hermaphrodite
      </label>
      <input
        checked={false}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.sex-unspecified"
        name="occurrenceForm.basicData.sex"
        onChange={[Function]}
        type="radio"
        value="unspecified"
      />
      <label
        htmlFor="occurrenceForm.basicData.sex-unspecified"
      >
        occurrenceForm.basicData.sex.unspecified
      </label>
    </div>
  </div>
  <div
    className="input-radio-group field"
  >
    <label
      className="label"
    >
      occurrenceForm.basicData.lifestage.title
    </label>
    <div
      className="control"
    >
      <input
        checked={true}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.lifestage-egg"
        name="occurrenceForm.basicData.lifestage"
        onChange={[Function]}
        type="radio"
        value="egg"
      />
      <label
        htmlFor="occurrenceForm.basicData.lifestage-egg"
      >
        occurrenceForm.basicData.lifestage.egg
      </label>
      <input
        checked={false}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.lifestage-eft"
        name="occurrenceForm.basicData.lifestage"
        onChange={[Function]}
        type="radio"
        value="eft"
      />
      <label
        htmlFor="occurrenceForm.basicData.lifestage-eft"
      >
        occurrenceForm.basicData.lifestage.eft
      </label>
      <input
        checked={false}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.lifestage-juvenile"
        name="occurrenceForm.basicData.lifestage"
        onChange={[Function]}
        type="radio"
        value="juvenile"
      />
      <label
        htmlFor="occurrenceForm.basicData.lifestage-juvenile"
      >
        occurrenceForm.basicData.lifestage.juvenile
      </label>
      <input
        checked={false}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.lifestage-adult"
        name="occurrenceForm.basicData.lifestage"
        onChange={[Function]}
        type="radio"
        value="adult"
      />
      <label
        htmlFor="occurrenceForm.basicData.lifestage-adult"
      >
        occurrenceForm.basicData.lifestage.adult
      </label>
      <input
        checked={false}
        className="input-radio is-checkradio"
        id="occurrenceForm.basicData.lifestage-unspecified"
        name="occurrenceForm.basicData.lifestage"
        onChange={[Function]}
        type="radio"
        value="unspecified"
      />
      <label
        htmlFor="occurrenceForm.basicData.lifestage-unspecified"
      >
        occurrenceForm.basicData.lifestage.unspecified
      </label>
    </div>
  </div>
  <div
    className="columns"
  >
    <div
      className="copy-previous-data column is-narrow"
    >
      copy data from previous entry
    </div>
  </div>
</div>
`;
